op-dispute-mon/mon/types/types.go

```diff
@@ -59,13 +59,51 @@ type EnrichedGameData struct {
 	// This ETH balance will be used to pay out any bonds required by the games
 	// that use the same DelayedWETH contract.
 	ETHCollateral *big.Int
+
+	// RollupEndpointErrors stores endpoint IDs that returned errors other than "not found" for this game.
+	RollupEndpointErrors map[string]bool
+
+	// RollupEndpointErrorCount tracks the total number of errors for this game across all endpoints.
+	RollupEndpointErrorCount int
+
+	// RollupEndpointNotFoundCount tracks the number of endpoints that returned "not found" for this game.
+	RollupEndpointNotFoundCount int
+
+	// RollupEndpointTotalCount tracks the total number of rollup endpoints attempted for this game.
+	RollupEndpointTotalCount int
+
+	// RollupEndpointSafeCount tracks the number of rollup endpoints that reported the root as safe.
+	RollupEndpointSafeCount int
+
+	// RollupEndpointUnsafeCount tracks the number of rollup endpoints that reported the root as unsafe.
+	RollupEndpointUnsafeCount int
+
+	// RollupEndpointDifferentOutputRoots tracks whether rollup endpoints returned different output roots for this game.
+	RollupEndpointDifferentOutputRoots bool
 }
 
 // UsesOutputRoots returns true if the game type is one of the known types that use output roots as proposals.
 func (g EnrichedGameData) UsesOutputRoots() bool {
 	return slices.Contains(outputRootGameTypes, g.GameType)
 }
 
+// HasMixedAvailability returns true if some rollup endpoints returned "not found" while others succeeded
+// for this game. This indicates inconsistent block availability across the rollup node network.
+func (g EnrichedGameData) HasMixedAvailability() bool {
+	if g.RollupEndpointTotalCount == 0 {
+		return false
+	}
+
+	successfulEndpoints := g.RollupEndpointTotalCount - g.RollupEndpointErrorCount - g.RollupEndpointNotFoundCount
+	return g.RollupEndpointNotFoundCount > 0 && successfulEndpoints > 0
+}
+
+// HasMixedSafety returns true if some rollup endpoints reported the root as safe and others as unsafe
+// for this game. This indicates inconsistent safety assessment across the rollup node network.
+func (g EnrichedGameData) HasMixedSafety() bool {
```

Comment:
Was this supposed to be recorded in a metric? I only see this function used by tests.